### PR TITLE
fix: import cycle not allowed

### PR DESCRIPTION
**Pull request by proletarius101**
The build log looks like this: https://download.copr.fedorainfracloud.org/results/proletarius101/hydroxide/fedora-32-x86_64/01654615-golang-github-emersion-message-textproto/builder-live.log.gz

I'd say I'm not quite familiar with Go. And as I build the source code manually with the same command there is no error:

```shell
go test -buildmode pie -compiler gc -ldflags " -X github.com/emersion/go-message/textproto/version.commit=master -X github.com/emersion/go-message/textproto/version=0 -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '"
```

I just made this commit by some guessing, and it works somehow.

### CODE REVIEW

**Comment by emersion** [comment]
```
can't load package: import cycle not allowed
package github.com/emersion/go-message/textproto
	imports github.com/emersion/go-message/textproto
```

This is a weird error. I'd recommend getting in touch with the Fedora Go maintainers. I suspect your Go package infrastructure is doing something weird.

In any case, I don't see how renaming a file can fix the issue.

**Comment by proletarius101** [comment]
OK, I'll get in touch with them. Anyway, hopefully I'm about to make hydroxide a Fedora package.

**Comment by proletarius101** [comment]
Bug filed to the upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1877760

**Comment by proletarius101** [comment]
So it's due to Fedora's `GO111MODULE=off`. Any hint why that's related to circle deps?

**Comment by proletarius101** [comment]
This should fix the problem. But the linter warns :(

**Comment by proletarius101** [comment]
Well, then that doens't comply with the official example: https://blog.golang.org/examples. Quite wired. Any ideas?

**Comment by emersion** [comment]
I don't get it. The only requirement on file names described in the blog post is:

> As with typical tests, examples are functions that reside in a package's _test.go files.

Which is the case before this PR...

And note the package in the blog post has the `_test` suffix, just like go-message. This is important so that examples show up with the right package name (users will need to write `textproto.Header`, not just `Header`).

tl;dr I don't want to degrade the quality of my package for Fedora packaging.

**Comment by proletarius101** [comment]
> tl;dr I don't want to degrade the quality of my package for Fedora packaging.

That's for sure. The problem here is, with `GO111MODULE=off`, the it doesn't builds, which means somehow the code is invalid before Go 1.11. So I'm wondering if there is any syntax error, or it's simply because some module changes after go 1.11?

**Comment by emersion** [comment]
Only Go >= 1.13 is supported: https://github.com/emersion/go-message/blob/e1abe21f389af315899bdb42c985995590503588/go.mod#L3

**Comment by proletarius101** [comment]
> Only Go >= 1.13 is supported

I mean yes they use Go 1.15. So is there any changes on module that related to this issue? If yes, I may have to convince them to drop that flag.

**Comment by emersion** [comment]
No, modules shouldn't change anything, provided you're using `GOPATH` correctly.

**Comment by proletarius101** [comment]
Since [go-rpm-macros](https://src.fedoraproject.org/rpms/go-rpm-macros) used in Fedora's build system hasn't support modules and won't support it in the near future, I decide to go through the easiest way(suggested by the [doc](https://docs.fedoraproject.org/en-US/packaging-guidelines/Golang_templates/#_minimal_binary)):

```
# When you package an import path, that participates in a dependency loop, you
# need bootstraping to manage the initial builds:
# https://docs.fedoraproject.org/en-US/packaging-guidelines/#bootstrapping
# For Go code, that means your bootstrap section should:
#  – remove unit tests that import other parts of the loop
#  – remove code that imports other parts of the loop
```

That is, remove unit test files leading to the problem beforehand:

```specfile
%prep
%goprep
# workaround cicle deps
for test_file in $(find . -name "*_test.go")
    do
        rm $test_file                                                                                                               
    done
```
